contentgrid-spring-querydsl/src/test/java/com/contentgrid/spring/querydsl/predicate/DefaultTest.java

```diff
@@ -0,0 +1,106 @@
+package com.contentgrid.spring.querydsl.predicate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.contentgrid.spring.querydsl.mapping.UnsupportedCollectionFilterPredicateException;
+import com.contentgrid.spring.querydsl.test.fixtures.QTestObject;
+import com.contentgrid.spring.querydsl.test.predicate.PredicateFactoryTester;
+import com.querydsl.core.types.Path;
+import com.querydsl.core.types.PathMetadataFactory;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.core.convert.support.ConfigurableConversionService;
+import org.springframework.core.convert.support.DefaultConversionService;
+import org.springframework.data.convert.Jsr310Converters;
+
+class DefaultTest {
+    private final Default DEFAULT_FACTORY = new Default();
+
+    private final PredicateFactoryTester<QTestObject> TESTER = new PredicateFactoryTester<>(new QTestObject(PathMetadataFactory.forVariable("o")));
+
+    private static final ConfigurableConversionService conversionService = new DefaultConversionService();
+    static {
+        Jsr310Converters.getConvertersToRegister().forEach(conversionService::addConverter);
```

Comment:
why do we need this ? the javadoc says it is only needed if you run on java 8 ?

Reply:
That javadoc is [from 9 years ago (2014)](https://github.com/spring-projects/spring-data-commons/blame/d0ba1252687acf416c838cef840b4145ab859814/src/main/java/org/springframework/data/convert/Jsr310Converters.java#L41), which is around the time that Java 8 was released. I guess it was used to avoid runtime class not found errors when running on Java 7 or earlier.

In fact, the javadoc was [updated recently](https://github.com/spring-projects/spring-data-commons/commit/dc999a81937bf05f2624fb5919e36b77027f3c22) to remove the Java 8 reference.